File: katas/FizzBuzz/solutions/fricke_frederik/CSharp/FizzBuzz/FizzBuzz/FizzBuzz.cs
namespace FizzBuzz;

public class FizzBuzz
{
    /// <summary>
    /// This is the primary function which checks if the given number is
    /// Fizz, Buzz or FizzBuzz
    /// </summary>
    /// <param name="numToCheck">Number to check for FizzBuzz</param>
    /// <returns>Returns a string which has Fizz, Buzz or FizzBuzz</returns>
    public string IsFizzBuzz(int numToCheck)
    {
        string outString = "";


        if (numToCheck % 3 == 0) outString += "Fizz";
        if (numToCheck % 5 == 0) outString += "Buzz";

        outString = CheckString(numToCheck, outString);

        if (outString == "") outString = numToCheck.ToString();
        
        return outString;
    }

    /// <summary>
    /// A helper function used to search if the given number has
    /// the number 3 or 5 in it.
    /// </summary>
    /// <param name="numToCheck">The given number which shall be checked</param>
    /// <param name="outString">A string which will returned with Fizz, Buzz or FizzBuzz</param>
    /// <param name="prevFizzBuzzString">the previous FizzBuzz-String</param>
    private string CheckString(int numToCheck, string prevFizzBuzzString)
    {
        string intString = numToCheck.ToString();
        string outString = prevFizzBuzzString;

        if (intString.Contains('3') && !(outString.Contains("Fizz"))) outString = "Fizz" + outString;
        if (intString.Contains('5') && !(outString.Contains("Buzz"))) outString += "Buzz";

        return outString;
    }
}

File: katas/FizzBuzz/solutions/fricke_frederik/Java/src/de/fricke/FizzBuzz.java
package de.fricke;

public class FizzBuzz {

    public String isFizzBuzz(int numToCheck)
    {
        String outString = "";

        if (numToCheck % 3 == 0) outString += "Fizz";
        if (numToCheck % 5 == 0) outString += "Buzz";

        outString = CheckString(numToCheck, outString);

        if (outString.equals("")) outString = Integer.toString(numToCheck);

        return outString;
    }

    private String CheckString(int numToCheck, String prevFizzBuzzString)
    {
        String outString = prevFizzBuzzString;
        String numString = Integer.toString(numToCheck);

        if (numString.contains("3") && !(outString.contains("Fizz"))) outString = "Fizz" + outString;
        if (numString.contains("5") && !(outString.contains("Buzz"))) outString += "Buzz";

        return outString;
    }
}
